bench_test.go

```diff
@@ -48,25 +48,122 @@ func BenchmarkConnect(b *testing.B) {
 	assert.True(b, ok)
 	httpTransport.DisableCompression = true
 
-	client := pingv1connect.NewPingServiceClient(
-		httpClient,
-		server.URL,
-		connect.WithGRPC(),
-		connect.WithSendGzip(),
-	)
+	clients := []struct {
+		name string
+		opts []connect.ClientOption
+	}{{
+		name: "connect",
+		opts: []connect.ClientOption{
+			connect.WithSendGzip(),
+		},
+	}, {
+		name: "grpc",
+		opts: []connect.ClientOption{
+			connect.WithSendGzip(),
+			connect.WithGRPC(),
+		},
+	}, {
+		name: "grpcweb",
+		opts: []connect.ClientOption{
+			connect.WithSendGzip(),
+			connect.WithGRPCWeb(),
+		},
+	}}
+
 	twoMiB := strings.Repeat("a", 2*1024*1024)
-	b.ResetTimer()
+	for _, client := range clients {
+		b.Run(client.name, func(b *testing.B) {
+			client := pingv1connect.NewPingServiceClient(
+				httpClient,
+				server.URL,
+				client.opts...,
+			)
 
-	b.Run("unary", func(b *testing.B) {
-		b.RunParallel(func(pb *testing.PB) {
-			for pb.Next() {
-				_, _ = client.Ping(
-					context.Background(),
-					connect.NewRequest(&pingv1.PingRequest{Text: twoMiB}),
-				)
-			}
+			ctx := context.Background()
+			b.Run("unary_big", func(b *testing.B) {
+				b.ReportAllocs()
+				b.RunParallel(func(pb *testing.PB) {
+					for pb.Next() {
+						_, err := client.Ping(
+							ctx, connect.NewRequest(&pingv1.PingRequest{Text: twoMiB}),
+						)
+						assert.Nil(b, err)
```

Comment:
Under stress benchmarks can fail, maybe theres an underlying race. If we don't check it'll always be in the back of my mind that it could be passing without validating. I'll remove the assert if it's only allocations your concerned with.